lib/buffer.js

```diff
@@ -561,14 +556,9 @@ Buffer.prototype.write = function(string, offset, length, encoding) {
       encoding = length;
       length = undefined;
     }
-
-  // XXX legacy write(string, encoding, offset, length) - remove in v0.13
   } else {
-    writeWarned = internalUtil.printDeprecationMessage(writeMsg, writeWarned);
-    var swap = encoding;
-    encoding = offset;
-    offset = length >>> 0;
-    length = swap;
+    throw new Error('Buffer.write(string, encoding, offset[, length]) ' +
```

Comment:
technically should be ``Buffer.write(string, encoding[, offset[, length]])'`, but I really don't care whether it's changed.